src/EditorFeatures/Core/Implementation/ChangeSignature/AbstractChangeSignatureCommandHandler.cs

```diff
@@ -68,25 +72,27 @@ private bool ExecuteCommand(ITextView textView, ITextBuffer subjectBuffer, Comma
                     return false;
                 }
 
-                var reorderParametersService = document.GetLanguageService<AbstractChangeSignatureService>();
-                var result = reorderParametersService.ChangeSignature(
+                var reorderParametersService = document.GetRequiredLanguageService<AbstractChangeSignatureService>();
+                var result = reorderParametersService.ChangeSignatureAsync(
                     document,
                     caretPoint.Value.Position,
-                    (errorMessage, severity) =>
-                    {
-                        // We are about to show a modal UI dialog so we should take over the command execution
-                        // wait context. That means the command system won't attempt to show its own wait dialog 
-                        // and also will take it into consideration when measuring command handling duration.
-                        context.OperationContext.TakeOwnership();
-                        workspace.Services.GetService<INotificationService>().SendNotification(errorMessage, severity: severity);
-                    },
-                context.OperationContext.UserCancellationToken);
+                    context.OperationContext.UserCancellationToken).WaitAndGetResult(context.OperationContext.UserCancellationToken);
 
+                var notificationService = workspace.Services.GetRequiredService<INotificationService>();
                 if (result == null || !result.Succeeded)
```

Comment:
is result null when we cancel? if so, doc that.

Reply:
it's nullable annotated to not return null - removed the check